packages/plugin/vite/src/util/plugins.ts

```diff
@@ -33,3 +35,48 @@ export function externalBuiltins() {
     },
   };
 }
+
+/**
+ * Hot restart App during development for better DX.
+ */
+export function hotRestart(
+  config: VitePluginBuildConfig,
+  // For `VitePluginBuildConfig['restart']` callback args.
+  context: { renderers: RendererConfigWithServer[] }
+) {
+  const restart = () => {
```

Comment:
suggestion: I think it would be good if this `restart()` helper was passed as a callback parameter to the custom `restart` function in the config.

usage example:
```js

          {
            // `entry` is just an alias for `build.lib.entry` in the corresponding file of `config`.
            entry: 'src/main.js',
            config: 'vite.main.config.mjs',
            restart: (rs) => {
              console.log('do some work in the custom restart hook');
              rs();
            }
          },
```

Reply:
Okay! But the  Preload scripts `reload()` still needed? 👀